src/op_analytics/cli/subcommands/pulls/defillama.py

```diff
@@ -88,60 +83,111 @@ def process_breakdown_stables(data):
         "price",
     ]
 
-    metadata = {}
+    metadata: Dict[str, Optional[str]] = {}
+
+    # Collect required metadata fields
     for key in must_have_metadata_fields:
+        if key not in data:
+            raise KeyError(f"Missing required metadata field: '{key}'")
         metadata[key] = data[key]
 
+    # Collect additional optional metadata fields
     for key in metadata_fields:
         metadata[key] = data.get(key)
 
-    result = pl.DataFrame(rows, infer_schema_length=len(rows)).with_columns(
-        id=pl.lit(metadata["id"]),
-        name=pl.lit(metadata["name"]),
-        symbol=pl.lit(metadata["symbol"]),
+    result = (
+        pl.DataFrame(rows, infer_schema_length=len(rows)).with_columns(
+            id=pl.lit(metadata["id"]),
+            name=pl.lit(metadata["name"]),
+            symbol=pl.lit(metadata["symbol"]),
+        )
+        if rows
+        else pl.DataFrame()
     )
 
     return result, metadata
 
 
-def pull_stables():
-    """Pull stablecoin data from DeFiLlama.
+def pull_stables(
+    stablecoin_ids: Optional[List[str]] = None, days: int = 30
+) -> Dict[str, pl.DataFrame]:
+    """
+    Pulls and processes stablecoin data from DeFiLlama.
+
+    Args:
+        stablecoin_ids: List of stablecoin IDs to process. Defaults to None (process all).
+        days: Number of days of data to retrieve. Defaults to 30.
 
-    - Fetch the stablecoins summary endpoint.
-    - For each stablecoin, fetch the detailed breakdown data.
-    - Write all results to BigQuery.
+    Returns:
+        A dictionary containing metadata and breakdown DataFrames.
     """
     session = new_session()
     summary = get_data(session, SUMMARY_ENDPOINT)
 
+    if "peggedAssets" not in summary:
```

Comment:
In the next line we do have `summary["peggedAssets"]` which will raise the KeyError if `peggedAssets` is not there. In the exception traceback we can see the line of code where the exception came from and that should give us the right context to understand the error.  I think it's ok to rely on system exceptions for the most data shape checks.  Or else we can end up reimplementing a lot of checks.   For example, what would happen if `summary["peggedAssets"]` is not an iterable.